src/main/java/com/mongodb/hibernate/jdbc/MongoPreparedStatement.java

```diff
@@ -312,4 +371,71 @@ private void checkParameterIndex(int parameterIndex) throws SQLException {
                     parameterIndex, parameterValueSetters.size()));
         }
     }
+
+    private int[] doExecuteBatch() throws SQLException {
+        startTransactionIfNeeded();
+        try {
+            var writeModels = new ArrayList<WriteModel<BsonDocument>>(commandBatch.size());
+            String collectionName = null;
+            for (BsonDocument command : commandBatch) {
+                var commandName = command.getFirstKey();
```

Comment:
Just so I understand correctly.  In Hibernate usage scenarios, is it the case that the command name will be the same for all commands in the batch?

Reply:
From current understanding, we can only guarantee that the collection or table name is the same (Hibernate groups statements internally), but it seems we can't assume the command names are the same.

Reply:
Yeah, by default Hibernate will group by the combination of the following two entries:

- entity name (without hierachy it maps to table or collection name)
- mutation type

So I could add another validation rule here as I've done for collection name; though seems not mandatory as the collection uniqueness validation.

Reply:
No need.  Just checking my mental model.

Reply:
simplified